qa/tasks/deepsea.py

```diff
@@ -1374,20 +1374,20 @@ def begin(self):
         # FIXME: this should be run only once - check for that and
         # return an error otherwise
         if self.munge_policy:
+            self.log.debug(self.munge_policy)
             for k, v in self.munge_policy.items():
-                if k == 'remove_storage_only_node':
-                    delete_me = self.first_storage_only_node()
-                    if not delete_me:
-                        raise ConfigError(
-                            self.err_prefix + "remove_storage_only_node "
-                            "requires a storage-only node, but there is no such"
+                if k in ['add_node', 'del_node']:
+                    try:
+                        teuthology_role, deepsea_role = v.items()[0]
+                    except AttributeError: # case of delete node
+                        teuthology_role = v
+                        deepsea_role = None
+                    remote = get_remote_for_role(self.ctx, teuthology_role)
+                    self.scripts.run(
+                            self.master_remote,
+                            'munge_policy.sh',
+                            args=[proposals_dir+"/policy.cfg", remote.hostname, k, deepsea_role]
                             )
```

Comment:
The `try..except` is confusing for me. Please consider the following alternative implementation:

```
teuthology_role = None
deepsea_role = None
if k in ['add_node', 'del_node']:
    try:
        teuthology_role, deepsea_role = v.items()[0]
    except AttributeError:
        raise ConfigError(
            self.err_prefix + "wrong configuration for {}".format(k)
            )
```

(the rest would be the same as now - in the `del_node` case, `deepsea_role` will be None/empty.)

Does that make sense?

Reply:
Yes